### PR TITLE
feat: add approval target

### PR DESCRIPTION
**Pull request by douglaseggleton**
# Description

Approval urls will occasionally have a hard requirement to continue the transaction in a blank window. This will only be returned for the payment methods that need it. The terminology of 'target' makes more sense to echo that of `self|blank` in a browser link. This also means we don't have to have context of Embed to interpret what a 'approval mode' means.

Screenshot 2022-05-13 at 09 34 34

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own changes
- [ ] I have run `yarn test` to make sure my changes pass all linters
- [ ] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/resources/payment_service_definition.yaml [hunk 1]
```diff
@@ -162,3 +162,16 @@ properties:
...
+  configuration:
+    type: object
+    nullable: true
+    description: |-
+      Congfiguration items for the payment service.
+    properties:
+      approval_ui_target:
+        type: string
+        nullable: true
+        description: Target than an approval URL must be opened in.
```
Comment (thomas-gr4vy) [inline reference/resources/payment_service_definition.yaml:175]:
```
The optional target that an approval URL must be opened in.
If not provided, there is no specific requirement.
```

File: reference/resources/payment_method.yaml [hunk 2]
```diff
@@ -108,9 +108,8 @@ properties:
     type: string
     example: blank
     nullable: true
-    description:
-      The optional target than an approval URL must be opened in. If not provided, there is no specific
-      requirement.
+    description: The optional target than an approval URL must be opened in.
```
Comment (thomas-gr4vy) [inline reference/resources/payment_method.yaml:111]: **Typo:** The optional target ~than~ that an approval URL must be opened in.

File: reference/resources/payment_method--tokenized.yaml [hunk 3]
```diff
@@ -66,9 +66,8 @@ properties:
     type: string
     example: blank
     nullable: true
-    description:
-      The optional target than an approval URL must be opened in. If not provided, there is no specific
-      requirement.
+    description: The optional target than an approval URL must be opened in.
```
Comment (thomas-gr4vy) [inline reference/resources/payment_method--tokenized.yaml:69]: **Typo:** The optional target ~than~ that an approval URL must be opened in.

File: reference/resources/payment_method--snapshot.yaml [hunk 4]
```diff
@@ -67,9 +67,8 @@ properties:
     type: string
     example: blank
     nullable: true
-    description:
-      The optional target than an approval URL must be opened in. If not provided, there is no specific
-      requirement.
+    description: The optional target than an approval URL must be opened in.
```
Comment (thomas-gr4vy) [inline reference/resources/payment_method--snapshot.yaml:70]: **Typo:** The optional target ~than~ that an approval URL must be opened in.

**Review by thomas-gr4vy — CHANGES_REQUESTED** [review]

File: reference/resources/payment_method.yaml [hunk 5]
```diff
@@ -104,6 +104,15 @@ properties:
...
     nullable: true
 
+  approval_target:
+    type: string
+    example: blank
+    nullable: true
+    description: The optional target than an approval URL must be opened in.
+      If not provided, there is no specific requirement.
+    enum:
+      - blank
```
Comment (cbetta) [inline reference/resources/payment_method.yaml:114]: We seem to re-use this prop all over. Might be easier to make a ref.

For example

```suggestion
  approval_target:
    $ref: "../properties/approval_target.yaml"
```

File: reference/resources/payment_method--snapshot.yaml [hunk 6]
```diff
@@ -63,6 +63,15 @@ properties:
...
 
+  approval_target:
+    type: string
+    example: blank
+    nullable: true
+    description: The optional target than an approval URL must be opened in.
+      If not provided, there is no specific requirement.
+    enum:
+      - blank
+
```
[the remark this answers is not in the document]
Comment (douglaseggleton) [reply reference/resources/payment_method--snapshot.yaml]: Discussed in standup - We are going to implement as replacing `blank` with `new_window` but not include additional values for now.

File: reference/resources/payment_method--snapshot.yaml [hunk 7]
```diff
@@ -63,6 +63,15 @@ properties:
     maxLength: 5
     nullable: true
 
+  approval_target:
+    type: string
+    example: blank
```
Comment (douglaseggleton) [inline reference/resources/payment_method--snapshot.yaml]:
```suggestion
    example:  "new_window"
```

File: reference/resources/payment_method--snapshot.yaml [hunk 8]
```diff
@@ -63,6 +63,15 @@ properties:
...
+    nullable: true
+    description: The optional target than an approval URL must be opened in.
+      If not provided, there is no specific requirement.
+    enum:
+      - blank
```
Comment (douglaseggleton) [inline reference/resources/payment_method--snapshot.yaml]:
```suggestion
      - "new_window"
```

File: reference/properties/approval_target.yaml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+---
+type: string
+example: blank
```
Comment (douglaseggleton) [inline reference/properties/approval_target.yaml:3]:
```suggestion
example: "new_window"
```

File: reference/properties/approval_target.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+nullable: true
+description: The optional target that an approval URL must be opened in.
+  If not provided, there is no specific requirement.
+enum:
+  - blank
```
Comment (douglaseggleton) [inline reference/properties/approval_target.yaml:8]:
```suggestion
  - "new_window"
```

File: reference/properties/approval_target.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+---
+type: string
+example: "new_window"
+nullable: true
+description: The optional target that an approval URL must be opened in.
+  If not provided, there is no specific requirement.
+enum:
+  - "new_window"
```
Comment (thomas-gr4vy) [inline reference/properties/approval_target.yaml:8]:
```suggestion
  - "new_window"
  - "any"
```

File: reference/properties/approval_target.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+---
+type: string
+example: any
+nullable: true
+description: The browser target that an approval URL must be opened in.
+  If not provided, there is no specific requirement.
```
Comment (thomas-gr4vy) [inline reference/properties/approval_target.yaml:6]: nitpick
```suggestion
  If `any` or `null`, then there is no specific requirement.
```

**Review by thomas-gr4vy — APPROVED** [review]

**Pull request merged**